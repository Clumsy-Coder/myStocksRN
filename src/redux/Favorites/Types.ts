import { Action } from 'redux';

/*
  ██████╗ ███████╗██████╗ ██╗   ██╗ ██████╗███████╗██████╗     ███████╗████████╗ █████╗ ████████╗███████╗
  ██╔══██╗██╔════╝██╔══██╗██║   ██║██╔════╝██╔════╝██╔══██╗    ██╔════╝╚══██╔══╝██╔══██╗╚══██╔══╝██╔════╝
  ██████╔╝█████╗  ██║  ██║██║   ██║██║     █████╗  ██████╔╝    ███████╗   ██║   ███████║   ██║   █████╗
  ██╔══██╗██╔══╝  ██║  ██║██║   ██║██║     ██╔══╝  ██╔══██╗    ╚════██║   ██║   ██╔══██║   ██║   ██╔══╝
  ██║  ██║███████╗██████╔╝╚██████╔╝╚██████╗███████╗██║  ██║    ███████║   ██║   ██║  ██║   ██║   ███████╗
  ╚═╝  ╚═╝╚══════╝╚═════╝  ╚═════╝  ╚═════╝╚══════╝╚═╝  ╚═╝    ╚══════╝   ╚═╝   ╚═╝  ╚═╝   ╚═╝   ╚══════╝

*/

/**
 * Favorites reducer state
 */
export interface FavoritesReducerState {
  symbols: string[];
}

// ---------------------------------------------------------------------------------------------- //
/*
   █████╗  ██████╗████████╗██╗ ██████╗ ███╗   ██╗    ████████╗██╗   ██╗██████╗ ███████╗███████╗
  ██╔══██╗██╔════╝╚══██╔══╝██║██╔═══██╗████╗  ██║    ╚══██╔══╝╚██╗ ██╔╝██╔══██╗██╔════╝██╔════╝
  ███████║██║        ██║   ██║██║   ██║██╔██╗ ██║       ██║    ╚████╔╝ ██████╔╝█████╗  ███████╗
  ██╔══██║██║        ██║   ██║██║   ██║██║╚██╗██║       ██║     ╚██╔╝  ██╔═══╝ ██╔══╝  ╚════██║
  ██║  ██║╚██████╗   ██║   ██║╚██████╔╝██║ ╚████║       ██║      ██║   ██║     ███████╗███████║
  ╚═╝  ╚═╝ ╚═════╝   ╚═╝   ╚═╝ ╚═════╝ ╚═╝  ╚═══╝       ╚═╝      ╚═╝   ╚═╝     ╚══════╝╚══════╝

*/

/**
 * Favorites action types
 */
export enum ActionTypes {
  ADD_FAVORITE_STOCK = 'FAVORITES/ADD_FAVORITE_STOCK',
  REMOVE_FAVORITE_STOCK = 'FAVORITES/REMOVE_FAVORITE_STOCK',
  CLEAR_FAVORITE_STOCKS = 'FAVORITES/CLEAR_FAVORITE_STOCKS',
}

// ---------------------------------------------------------------------------------------------- //
/*
   █████╗  ██████╗████████╗██╗ ██████╗ ███╗   ██╗███████╗
  ██╔══██╗██╔════╝╚══██╔══╝██║██╔═══██╗████╗  ██║██╔════╝
  ███████║██║        ██║   ██║██║   ██║██╔██╗ ██║███████╗
  ██╔══██║██║        ██║   ██║██║   ██║██║╚██╗██║╚════██║
  ██║  ██║╚██████╗   ██║   ██║╚██████╔╝██║ ╚████║███████║
  ╚═╝  ╚═╝ ╚═════╝   ╚═╝   ╚═╝ ╚═════╝ ╚═╝  ╚═══╝╚══════╝

*/

/**
 * Add stock to Favorites action
 */
export interface AddFavoriteStockAction extends Action<ActionTypes.ADD_FAVORITE_STOCK> {
  type: ActionTypes.ADD_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Remove stock from Favorites action
 */
export interface RemoveFavoriteStockAction extends Action<ActionTypes.REMOVE_FAVORITE_STOCK> {
  type: ActionTypes.REMOVE_FAVORITE_STOCK;
  readonly stockSymbol: string;
}

/**
 * Clear all stocks from Favorites action
 */
export interface ClearFavoriteStocksAction extends Action<ActionTypes.CLEAR_FAVORITE_STOCKS> {
  type: ActionTypes.CLEAR_FAVORITE_STOCKS;
}

/**
 * Union of actions for AddFavoriteStock, RemoveFavoriteStock, ClearFavoriteStocks
 */
export type FavoritesActions =
  | AddFavoriteStockAction
  | RemoveFavoriteStockAction
  | ClearFavoriteStocksAction;
