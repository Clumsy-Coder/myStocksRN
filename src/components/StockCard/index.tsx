import React from 'react';
import { View, StyleSheet } from 'react-native';
import { ListItem, Text, H1, H3 } from 'native-base';
import { useNavigation } from '@react-navigation/native';

import { Selectors } from 'src/redux/Stocks/Types';
import { StockDetailsNavigationProp, NavigationRoutePath } from '@routes/Types';

interface OwnProps {
  data: Selectors.SelectQuoteTrim;
}

const styleSheet = StyleSheet.create({
  // main container
  container: {
    // backgroundColor: 'blue',
  },
  listItemContainer: {
    flex: 1,
    flexDirection: 'row',
    height: 70,
  },
  // company
  companyContainer: {
    flex: 3,
    flexDirection: 'column',
    // backgroundColor: 'orange',
    justifyContent: 'space-around',
  },
  companyStockSymbol: {
    fontWeight: 'bold',
  },
  companySymbol: {
    fontWeight: 'bold',
  },
  companyName: {},
  // price
  priceContainer: {
    flex: 1,
    flexDirection: 'column',
    // backgroundColor: 'pink',
    justifyContent: 'space-around',
    alignItems: 'flex-end',
  },
  priceValue: {},
  priceChangeContainer: {
    borderRadius: 4,
    padding: 1,
    paddingVertical: 2,
    paddingRight: 5,
    width: '100%',
    alignItems: 'flex-end',
  },
  priceChange: {
    color: 'white',
  },
  priceRed: {
    backgroundColor: 'red',
  },
  priceGreen: {
    backgroundColor: 'green',
  },
});

const StockCard: React.FC<OwnProps> = (props: OwnProps) => {
  const { data } = props;
  const navigation = useNavigation<StockDetailsNavigationProp>();

  if (data.fetching) {
    return (
      <ListItem>
        <Text>Loading</Text>
      </ListItem>
    );
  }

  const change = data.change === undefined ? '0' : data.change;

  return (
    <ListItem
      style={styleSheet.container}
      onPress={(): void =>
        navigation.navigate(NavigationRoutePath.StockDetails, { symbol: data.symbol })
      }
    >
      <View style={styleSheet.listItemContainer}>
        <View style={styleSheet.companyContainer}>
          <H1 style={styleSheet.companySymbol}>{data.symbol}</H1>
          <H3 style={styleSheet.companyName}>{data.companyName}</H3>
        </View>
        <View style={styleSheet.priceContainer}>
          <H3>{data.price.toFixed(2)}</H3>
          <View
            style={[
              styleSheet.priceChangeContainer,
              +change > 0 ? styleSheet.priceGreen : styleSheet.priceRed,
            ]}
          >
            <H3 style={styleSheet.priceChange}>{data.change}</H3>
          </View>
        </View>
      </View>
    </ListItem>
  );
};

export default StockCard;
