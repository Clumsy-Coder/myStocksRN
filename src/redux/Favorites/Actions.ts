import { ActionCreator } from 'redux';

import {
  ActionTypes,
  AddFavoriteStockAction,
  RemoveFavoriteStockAction,
  ClearFavoriteStocksAction,
} from 'src/redux/Favorites/Types';

/**
 * Add stock to Favorites action creator.
 * @param stockSymbol - Stock symbol in uppercase. Ex: AAPL
 */
export const addFavoriteStock: ActionCreator<AddFavoriteStockAction> = (
  stockSymbol: string,
): AddFavoriteStockAction => ({
  type: ActionTypes.ADD_FAVORITE_STOCK,
  stockSymbol,
});

/**
 * Remove stock from Favorites action creator.
 * @param stockSymbol - Stock symbol in uppercase. Ex: AAPL
 */
export const removeFavoriteStock: ActionCreator<RemoveFavoriteStockAction> = (
  stockSymbol: string,
): RemoveFavoriteStockAction => ({
  type: ActionTypes.REMOVE_FAVORITE_STOCK,
  stockSymbol,
});

/**
 * Clear all stocks from Favorites action creator.
 */
// eslint-disable-next-line max-len
export const clearFavoriteStocks: ActionCreator<ClearFavoriteStocksAction> = (): ClearFavoriteStocksAction => ({
  type: ActionTypes.CLEAR_FAVORITE_STOCKS,
});
