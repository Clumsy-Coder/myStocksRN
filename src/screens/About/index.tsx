import * as React from 'react';
import { Button, View, Text } from 'react-native';

import NavigationRoutePath from 'src/routes/RoutePath.enum';

const SettingScreen = ({ navigation }) => {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text>About screen</Text>
      <Button
        onPress={() => navigation.navigate(NavigationRoutePath.Settings)}
        title='Go to Settings screen'
      />
    </View>
  );
};

export default SettingScreen;
