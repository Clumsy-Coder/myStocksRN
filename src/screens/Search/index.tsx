import React from 'react';
import { Container, Content } from 'native-base';

import SearchBarHeader from '@containers/SearchBar';

const SearchScreen: React.FC = () => (
  <Container>
    <Content>
      <SearchBarHeader />
    </Content>
  </Container>
);

export default SearchScreen;
