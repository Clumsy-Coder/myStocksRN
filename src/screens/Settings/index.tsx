import * as React from 'react';
import { Button, View, Text } from 'react-native';
import { StackNavigationProp } from '@react-navigation/stack';

import NavigationRoutePath from 'src/routes/RoutePath.enum';
import { RootStackParamList } from 'src/routes/AppRouteTypes';

type Props = {
  navigation: StackNavigationProp<RootStackParamList, 'Settings'>;
};

const SettingScreen = ({ navigation }: Props) => {
  return (
    <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
      <Text>Setting screen</Text>
      <Button
        onPress={() => navigation.navigate(NavigationRoutePath.Home)}
        title='Go to Home screen'
      />
      <Button
        onPress={() => navigation.navigate(NavigationRoutePath.About)}
        title='Go to About screen'
      />
    </View>
  );
};

export default SettingScreen;
