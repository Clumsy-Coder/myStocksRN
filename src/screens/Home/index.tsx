import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { StackNavigationProp } from '@react-navigation/stack';
import { Container, Content, List } from 'native-base';

import { AppActions, AppState } from 'src/redux/index.reducers';
import { RootStackParamList } from 'src/routes/AppRouteTypes';
import { fetchStockQuoteBatch } from 'src/redux/Stocks/Actions';
import { Actions as StocksActions, Selectors as StocksSelectors } from 'src/redux/Stocks/Types';
import { selectStockQuoteTrim } from 'src/redux/Stocks/Selectors';
import StockCard from 'src/components/StockCard';

interface OwnProps {
  navigation: StackNavigationProp<RootStackParamList, 'Home'>;
}

interface SelectorProps {
  selectedStockQuoteTrim: StocksSelectors.SelectQuoteTrim[];
}

interface DispatchProps {
  fetchQuoteBatch: () => StocksActions.Batch.FetchQuoteAction;
}

type Props = SelectorProps & DispatchProps & OwnProps;

export class Home extends React.Component<Props> {
  componentDidMount(): void {
    const { fetchQuoteBatch } = this.props;
    fetchQuoteBatch();
  }

  render(): JSX.Element {
    const { selectedStockQuoteTrim } = this.props;

    return (
      <Container>
        <Content>
          <List>
            {selectedStockQuoteTrim.map((stockQuoteTrim: StocksSelectors.SelectQuoteTrim) => {
              return (
                <StockCard
                  data={stockQuoteTrim}
                  key={`stock-quote-trim-${stockQuoteTrim.symbol}`}
                />
              );
            })}
          </List>
        </Content>
      </Container>
    );
  }
}

const mapStateToProps = (state: AppState): SelectorProps => ({
  selectedStockQuoteTrim: selectStockQuoteTrim(state),
});

const mapDispatchToProps = (dispatch: Dispatch<AppActions>): DispatchProps => ({
  fetchQuoteBatch: (): StocksActions.Batch.FetchQuoteAction => dispatch(fetchStockQuoteBatch()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Home);
