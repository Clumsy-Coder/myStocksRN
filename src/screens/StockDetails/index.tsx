import * as React from 'react';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { Container, Content, Text, H1, View, Button, Spinner } from 'native-base';
import { Dispatch } from 'redux';

import { StockDetailsRouteProp } from '@routes/Types';
import { AppState, AppActions } from '@redux/index.reducers';
import { Actions, Selectors, DataDomain } from '@redux/Stocks/Types';
import { fetchStockChart } from '@redux/Stocks/Actions';
import { selectStockDetailsTrim, selectStockDetailsLineChart } from '@redux/Stocks/Selectors';
import { defaultQuote } from '@redux/Stocks/Reducer';
import StocksKeyStats from '@components/StocksKeyStats';
import LineChartView from '@components/LineChart';

interface SelectorProps {
  data: Selectors.SelectStockDetailsTrim;
  lineChartData: { date: string; price: number }[];
}

interface DispatchProps {
  fetchChart: (symbol: string, chartRange: DataDomain.ChartRange) => Actions.Chart.FetchAction;
}

type NavigationProps = {
  route: StockDetailsRouteProp;
};

type Props = NavigationProps & SelectorProps & DispatchProps;

type State = {
  chartRange: DataDomain.ChartRange;
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
    flexDirection: 'row',
    padding: 10,
  },
  price: {
    fontWeight: 'bold',
    fontSize: 40,
    paddingTop: 20,
  },
  chartRange: {
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'nowrap',
    justifyContent: 'space-between',
    paddingBottom: 10,
    flexGrow: 1,
    flexBasis: '100%',
  },
  loadingView: {
    flex: 1,
    justifyContent: 'center',
    alignContent: 'center',
    alignItems: 'center',
  },
});

export class StockDetailsScreen extends React.Component<Props, State> {
  private fetchChartInterval: NodeJS.Timeout | undefined;

  private fetchChartIntervalTimeout = 10;

  constructor(props: Props) {
    super(props);
    this.state = {
      chartRange: DataDomain.ChartRange.ONE_MONTH,
    };
  }

  componentDidMount = (): void => {
    const { route, fetchChart } = this.props;
    const { chartRange } = this.state;

    fetchChart(route.params.symbol, chartRange); // setInterval won't fetch chart. It will wait until time runs out.
    this.fetchChartInterval = setInterval(
      fetchChart.bind(this, route.params.symbol, chartRange),
      this.fetchChartIntervalTimeout * 1000,
    );
  };

  componentDidUpdate = (prevProps: Props, prevState: State): void => {
    const { chartRange } = this.state;

    if (prevState.chartRange !== chartRange) {
      const { route, fetchChart } = this.props;

      fetchChart(route.params.symbol, chartRange);

      this.fetchChartInterval = setInterval(
        fetchChart.bind(this, route.params.symbol, chartRange),
        this.fetchChartIntervalTimeout * 1000,
      );
    }
  };

  componentWillUnmount = (): void => {
    if (this.fetchChartInterval) clearInterval(this.fetchChartInterval);
  };

  onChartRangeChange = (newChartRange: DataDomain.ChartRange): void => {
    if (this.fetchChartInterval) clearInterval(this.fetchChartInterval);
    this.setState({ chartRange: newChartRange });
  };

  render(): JSX.Element {
    const { data, lineChartData } = this.props;
    const { chartRange } = this.state;

    const chartRangeList: DataDomain.ChartRange[] = [
      DataDomain.ChartRange.FIVE_DAYS,
      DataDomain.ChartRange.ONE_MONTH,
      DataDomain.ChartRange.SIX_MONTHS,
      DataDomain.ChartRange.YEAR_TO_DATE,
      DataDomain.ChartRange.ONE_YEAR,
      DataDomain.ChartRange.FIVE_YEARS,
      DataDomain.ChartRange.MAX,
    ];

    if (data.fetching && data.quote === defaultQuote && data.chart.length === 0) {
      return (
        <Container>
          <Content>
            <Spinner />
          </Content>
        </Container>
      );
    }

    return (
      <Container style={styles.root}>
        <Content>
          <View>
            <H1>{data.quote?.companyName}</H1>
          </View>
          <H1 style={styles.price}>{data.quote.latestPrice.toFixed(2)}</H1>
          <LineChartView data={lineChartData} />
          <View style={styles.chartRange}>
            {chartRangeList.map((cur) => (
              <Button
                key={`chartRange-${cur.toUpperCase()}`}
                bordered={chartRange === cur ? undefined : true}
                onPress={(): void => this.onChartRangeChange(cur)}
              >
                <Text>{cur.toUpperCase()}</Text>
              </Button>
            ))}
          </View>
          <StocksKeyStats data={data.quote} />
        </Content>
      </Container>
    );
  }
}

export const mapStateToProps = (state: AppState, ownProps: Props): SelectorProps => ({
  data: selectStockDetailsTrim(state, {
    stockSymbol: ownProps.route.params.symbol,
  }),
  lineChartData: selectStockDetailsLineChart(state, {
    stockSymbol: ownProps.route.params.symbol,
  }),
});

export const mapDispatchToProps = (dispatch: Dispatch<AppActions>): DispatchProps => ({
  fetchChart: (symbol: string, chartRange: DataDomain.ChartRange): Actions.Chart.FetchAction =>
    dispatch(fetchStockChart(symbol, chartRange)),
});

export default connect(mapStateToProps, mapDispatchToProps)(StockDetailsScreen);
