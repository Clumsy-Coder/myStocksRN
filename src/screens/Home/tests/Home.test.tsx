import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { Home, mapStateToProps, mapDispatchToProps } from 'src/screens/Home';
import StockCard from 'src/components/StockCard';
import { AppState } from 'src/redux/index.reducers';
import { Reducer, DataDomain, Selectors } from 'src/redux/Stocks/Types';
import { fetchStockQuoteBatch } from 'src/redux/Stocks/Actions';

import * as testdata from 'jest.testdata';

const configureMockStore = configureStore<AppState>();

describe('<Home />', () => {
  const selectedStockQuoteTrim: Selectors.SelectQuoteTrim[] = [
    {
      fetching: false,
      companyName: testdata.stockQuoteData1.companyName,
      symbol: testdata.stockSymbol1,
      price: testdata.stockQuoteData1.latestPrice,
      changePercent: testdata.stockQuoteData1.changePercent,
      change: testdata.stockQuoteData1.change,
      currency: testdata.symbolsMetadata1.currency,
    },
    {
      fetching: false,
      companyName: testdata.stockQuoteData2.companyName,
      symbol: testdata.stockSymbol2,
      price: testdata.stockQuoteData2.latestPrice,
      changePercent: testdata.stockQuoteData2.changePercent,
      change: testdata.stockQuoteData2.change,
      currency: testdata.symbolsMetadata2.currency,
    },
    {
      fetching: false,
      companyName: testdata.stockQuoteData3.companyName,
      symbol: testdata.stockSymbol3,
      price: testdata.stockQuoteData3.latestPrice,
      changePercent: testdata.stockQuoteData3.changePercent,
      change: testdata.stockQuoteData3.change,
      currency: testdata.symbolsMetadata3.currency,
    },
  ];

  const props = {
    selectedStockQuoteTrim,
    fetchQuoteBatch: jest.fn(),
  };

  const testID = 'stocks-home-list';

  it('Should render correctly', () => {
    const wrapper = shallow(<Home {...props} selectedStockQuoteTrim={selectedStockQuoteTrim} />);

    expect(wrapper).toMatchSnapshot();
  });

  it.skip('Should call fetchStockQuoteBatch in componentDidMount', () => {});

  it('Should render StockCard components', () => {
    const wrapper = shallow(<Home {...props} selectedStockQuoteTrim={selectedStockQuoteTrim} />);
    const stockList = wrapper.find({ testID });

    expect(stockList.children().length).toEqual(3);
    expect(stockList.childAt(0).equals(<StockCard data={selectedStockQuoteTrim[0]} />)).toEqual(
      true,
    );
    expect(stockList.childAt(1).equals(<StockCard data={selectedStockQuoteTrim[1]} />)).toEqual(
      true,
    );
    expect(stockList.childAt(2).equals(<StockCard data={selectedStockQuoteTrim[2]} />)).toEqual(
      true,
    );
  });

  describe('mapStateToProps', () => {
    it(`Property 'selectedStockQuoteTrim' works properly`, () => {
      const rootState: AppState = {
        ...testdata.baseAppState,
        Stocks: {
          ...testdata.baseStocksState,
          symbols: {
            [testdata.stockSymbol1]: {
              quote: {
                fetching: false,
                data: testdata.stockQuoteData1,
                error: undefined,
              },
            },
            [testdata.stockSymbol2]: {
              quote: {
                fetching: false,
                data: testdata.stockQuoteData2,
                error: undefined,
              },
            },
            [testdata.stockSymbol3]: {
              quote: {
                fetching: false,
                data: testdata.stockQuoteData3,
                error: undefined,
              },
            },
          },
          symbolsMetadata: {
            fetching: false,
            data: [testdata.symbolsMetadata1, testdata.symbolsMetadata2, testdata.symbolsMetadata3],
            error: undefined,
          },
        },
        Favorites: {
          symbols: [testdata.stockSymbol1, testdata.stockSymbol2, testdata.stockSymbol3],
        },
      };

      const stateToProps = mapStateToProps(rootState);

      expect(stateToProps).toHaveProperty('selectedStockQuoteTrim');
      expect(Array.isArray(stateToProps.selectedStockQuoteTrim)).toEqual(true);
      expect(stateToProps.selectedStockQuoteTrim.length).toEqual(3);
      expect(stateToProps.selectedStockQuoteTrim[0]).toEqual(selectedStockQuoteTrim[0]);
      expect(stateToProps.selectedStockQuoteTrim[1]).toEqual(selectedStockQuoteTrim[1]);
      expect(stateToProps.selectedStockQuoteTrim[2]).toEqual(selectedStockQuoteTrim[2]);
    });
  });

  describe('mapDispatchToProps', () => {
    it('fetchQuoteBatch works properly', () => {
      const dispatch = jest.fn();

      const dispatchToProps = mapDispatchToProps(dispatch);
      dispatchToProps.fetchQuoteBatch();

      expect(dispatch.mock.calls[0].length).toEqual(1);
      expect(dispatch.mock.calls[0][0]).toEqual(fetchStockQuoteBatch());
    });
  });
});
