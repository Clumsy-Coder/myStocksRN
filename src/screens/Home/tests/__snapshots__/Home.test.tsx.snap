// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Should render correctly 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <Styled(List)
      testID="stocks-home-list"
    >
      <StockCard
        data={
          Object {
            "change": -1.69,
            "changePercent": -1.361,
            "companyName": "International Business Machines Corp.",
            "currency": "USD",
            "fetching": false,
            "price": 122.47,
            "symbol": "IBM",
          }
        }
        key="stock-quote-trim-IBM"
      />
      <StockCard
        data={
          Object {
            "change": -2.18,
            "changePercent": -0.62,
            "companyName": "Apple, Inc.",
            "currency": "USD",
            "fetching": false,
            "price": 349.55,
            "symbol": "AAPL",
          }
        }
        key="stock-quote-trim-AAPL"
      />
      <StockCard
        data={
          Object {
            "change": 17.44,
            "changePercent": 2.02,
            "companyName": "Shopify, Inc.",
            "currency": "USD",
            "fetching": false,
            "price": 881,
            "symbol": "SHOP",
          }
        }
        key="stock-quote-trim-SHOP"
      />
    </Styled(List)>
  </Styled(Content)>
</Styled(Container)>
`;
